scipy/cluster/_hierarchy.pyx

```diff
@@ -135,7 +135,7 @@ def cluster_maxclust_dist(const double[:, :] Z, int[:] T, int n, int mc):
     mc : int
         The maximum number of clusters.
     """
-    cdef double[:] max_dists = np.ndarray(n, dtype=np.float64)
+    cdef double[:] max_dists = np.ndarray(n - 1, dtype=np.float64)
```

Comment:
This is also a minor bug, only first `n - 1` elements are actually used and required.